core/shared/src/main/scala/scala/macros/config/Api.scala

```diff
@@ -0,0 +1,18 @@
+package scala.macros
+package config
+
+import scala.macros.internal.config.scalamacros.BuildInfo
+
+private[macros] trait Api {
+  private def ensureVersion(key: String, value: String): Version = {
+    def fail = sys.error(s"fatal error reading BuildInfo: $key $value is not a valid version")
```

Comment:
Do we do this with the current paradise impl? This looks incredibly useful and I presume it would solve some issues we see in gitter.